Argo/Types/Decoded/Applicative.swift

```diff
@@ -45,9 +45,12 @@ public extension Decoded {
     - returns: A value of type `Decoded<U>`
   */
   func apply<U>(_ f: Decoded<(T) -> U>) -> Decoded<U> {
-    switch f {
-    case let .success(function): return self.map(function)
-    case let .failure(error): return .failure(error)
+    switch (f, self) {
+    case let (.success(function), _): return self.map(function)
+    case let (.failure(.multiple(es)), .failure(e)): return .failure(.multiple(es + [e]))
+    case let (.failure(e), .failure(.multiple(es))): return .failure(.multiple([e] + es))
+    case let (.failure(le), .failure(re)): return .failure(.multiple([le, re]))
+    case let (.failure(f), _): return .failure(f)
```

Comment:
Had a thought about this? What do you think about implementing `+` for `DecodeError` and using it here instead of matching on the specific error case?